cypress/support/components/EditDialogModal.js

```diff
@@ -42,10 +42,15 @@ export function VerifyTags(tags) {
   })
 }
 
-export function AddTag(tag) { 
-  cy.Get('button.cl-tags__button-add#tags').Click()
-  cy.Get('input#tags').type(`${tag}{enter}`)
-  cy.WaitForStableDOM()
+// Pass in either an array of tag strings or a single tag string.
+// If you try to call this twice in a row, it will fail to find the "Add Tag Button"
+// so don't do it, this was designed to take multiple tags.
+export function AddTags(tags) { 
+  ClickAddTagButton()
+  let tagList = ''
+  if (!Array.isArray(tags)) { tagList = `${tags}{enter}` }
```

Comment:
Doesn't look like it would solve the issue about switching types. E.g. string to array, it just changed the implementation depending on if it is a string or array. This now worse than before because it generates the`tagList` in two places conditionally.  I don't know how to make myself more clear:

> you should haven't change the type of arguments depending on whether you want to add 1 or more.

If at one point I write a test that adds 1 tag: `AddTags('MyTag')` and then I realize later it should add 2 tags I should NOT have to change the type. Meaning just add a tag: `AddTags('MyTag', 'MySecondTag')` The argument types stays as strings.

You either take multiple arguments as strings and array of strings.  Could do both but that's more complicated and would rather people use the method the same way every time.  The having mixed and people being confused.

Reply:
The other change I made was because I was thinking your main objection was changing the type of a function argument. Sorry I misunderstood. I fixed this. Need to re-review the code. Impacted many places.

I actually liked it the other way, it was like function/method overloading in other languages. I thought other people would like it too. Clearly you don't so it is gone. They now all take arrays, which I think is awkward/unnecessary when you pass in a single item as is the most common use case with these functions.